WordPress/gutenberg
# Editor: when Toggle navigation receives state false, focus

## Description

In the Site Editor, Toggle navigation button will have focus placed on the trigger when open state changes to false. This is so when screen reader/keyboard users select an item from the menu, they are dropped back at a global point "Toggle navigation" which is the first element on the page.

Follow up to: #36597

## How has this been tested?

I tested in Firefox with NVDA screen reader on Windows 10.

To test:
1. Activate a block based theme such as TT1 Blocks or Twenty-Twenty Two.
2. Open the Site Editor. Appearance > Editor. (Beta)
3. Select Toggle navigation.
4. Select one of the navigation items.
5. Notice how the sidebar closes and focus is placed back on Toggle navigation button.

## Screenshots 

## Types of changes





Bug fix.

## Checklist:
